android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -134,23 +138,30 @@ public void onClick(View v) {
 
     private void setUpAssetList(NetworkInfo selectedNetwork) {
         AccountInfo[] accountInfos = new AccountInfo[] {getThisAccountInfo()};
-        PortfolioHelper portfolioHelper = new PortfolioHelper(this, accountInfos);
-        portfolioHelper.setSelectedNetwork(selectedNetwork);
-        portfolioHelper.calculateBalances(() -> {
-            RecyclerView rvAssets = findViewById(R.id.rv_assets);
-
-            BlockchainToken[] userAssets = portfolioHelper.getUserAssets();
-            HashMap<String, Double> perTokenCryptoSum = portfolioHelper.getPerTokenCryptoSum();
-            HashMap<String, Double> perTokenFiatSum = portfolioHelper.getPerTokenFiatSum();
-
-            String tokensPath = BlockchainRegistryFactory.getInstance().getTokensIconsLocation();
-
-            WalletCoinAdapter walletCoinAdapter = Utils.setupVisibleAssetList(
-                    userAssets, perTokenCryptoSum, perTokenFiatSum, tokensPath);
-            walletCoinAdapter.setOnWalletListItemClick(AccountDetailActivity.this);
-            rvAssets.setAdapter(walletCoinAdapter);
-            rvAssets.setLayoutManager(new LinearLayoutManager(this));
-        });
+        LiveDataUtil.observeOnce(
+                mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks, allNetworks -> {
```

Comment:
null check on `mWalletModel`

Reply:
yeah, but we get that object like that right?
```
        BraveActivity activity = BraveActivity.getBraveActivity();
        if (activity != null) {
            mWalletModel = activity.getWalletModel();
        }
```
put an assert at least if you think it's impossible at all so we know that something is off while in debug mode.

Reply:
Also I checked other places using model, there are no null checks so I am not sure I understand?

Reply:
I've really seen that activity was null in the past, I didn't know how to replicate but it really happened to me on debug builds at least. That `if (activity != null) {` isn't just in case. We missed other places in that case.

Reply:
Ok I've added checks to all places. Also add some missing checks in `PortfolioFragment`.